azurerm_cdn_frontdoor_secret crashes provider v5.0.1 with panic: interface conversion: []interface {} is not map[string]interface {}

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.14.0

### AzureRM Provider Version

5.0.1

### Affected Resource(s)/Data Source(s)

azurerm_cdn_frontdoor_secret

### Affected Location(s)

westeurope

### Terraform Configuration Files

```hcl
resource "azurerm_cdn_frontdoor_secret" "secret" {
  for_each                 = { for item in var.custom_domains : item["name"] => item if var.enabled_certificate }
  name                     = "${each.key}-secret"
  cdn_frontdoor_profile_id = azurerm_cdn_frontdoor_profile.profile.id
  secret {
    customer_certificate {
      key_vault_certificate_id = "https://${each.value.certificate.key_vault_name}.vault.azure.net/certificates/${each.value.certificate.name}"
    }
  }
}

variable "custom_domains" {
  type = list(object({
    name : string
    host_name : string
    certificate : optional(object({
      name : string
      key_vault_name : string
      key_vault_resource_group : string
    }))
    routes : list(string)
  }))
  validation {
    condition     = alltrue([for domain in var.custom_domains : length(domain.routes) > 0])
    error_message = "Each custom domain must have at least one route defined in the routes list. Empty routes list is not allowed."
  }
  default     = []
  description = "The name which should be used for this Front Door Custom Domain and The host name of the domain. The host_name field must be the FQDN of your domain"
}
```

### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v5.0.1_x5 plugin:

panic: interface conversion: interface {} is []interface {}, not map[string]interface {}

goroutine 1479 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn.expandCdnFrontDoorSecretParameters({0x8c32d80, 0x344a1f33d6c0}, {0x344a2023c1b0?, 0x25?, 0x344a1f7f06ca?}, 0x344a20ff6908)
	github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/cdn_frontdoor_secret_resource.go:212 +0x5fb
github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn.resourceCdnFrontDoorSecretCreate(0x344a2180b7a0, {0x7621ec0, 0x344a20ff6908})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/cdn_frontdoor_secret_resource.go:132 +0x3f1
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x8c32cd8?, {0x8c32cd8?, 0x344a20eeec60?}, 0xd?, {0x7621ec0?, 0x344a20ff6908?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.40.1/helper/schema/resource.go:837 +0x15f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x344a1f814a00, {0x8c32cd8, 0x344a20eeec60}, 0x344a21debc70, 0x344a2180b680, {0x7621ec0, 0x344a20ff6908})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.40.1/helper/schema/resource.go:980 +0xafd
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x344a1f875aa0, {0x8c32cd8?, 0x344a20eeeba0?}, 0x344a1fd12cd0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.40.1/helper/schema/grpc_provider.go:1507 +0x10c9
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0x344a205b6a20, {0x8c32cd8?, 0x344a20eee8d0?}, 0x344a1fd12cd0)
	github.com/hashicorp/terraform-plugin-mux@v0.23.1/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x18d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x344a20c1a6e0, {0x8c32cd8?, 0x344a20a8f5f0?}, 0x344a21d0d100)
	github.com/hashicorp/terraform-plugin-go@v0.31.0/tfprotov5/tf5server/server.go:944 +0x3b9
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x86ef800, 0x344a20c1a6e0}, {0x8c32cd8, 0x344a20a8f5f0}, 0x344a21d0d080, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.31.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:805 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0x344a20bfeb48, {0x8c32cd8, 0x344a20a8f230}, 0x344a1fec36c0, 0x344a20213c50, 0xe98e968, 0x0)
	google.golang.org/grpc@v1.82.1/server.go:1443 +0x11dc
google.golang.org/grpc.(*Server).handleStream(0x344a20bfeb48, {0x8c36cb8, 0x344a20086820}, 0x344a1fec36c0)
	google.golang.org/grpc@v1.82.1/server.go:1858 +0xae6
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.82.1/server.go:1078 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 39
	google.golang.org/grpc@v1.82.1/server.go:1089 +0x11d

Error: The terraform-provider-azurerm_v5.0.1_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behaviour

_No response_

### Actual Behaviour

_No response_

### Steps to Reproduce

terraform apply

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/cdn/cdn_frontdoor_secret_resource.go
+++ b/internal/services/cdn/cdn_frontdoor_secret_resource.go
@@ -209,9 +209,10 @@ func resourceCdnFrontDoorSecretDelete(d *pluginsdk.ResourceData, meta interface{
 func expandCdnFrontDoorSecretParameters(ctx context.Context, input []interface{}, clients *clients.Client) (secrets.SecretParameters, error) {
 	v := input[0].(map[string]interface{})
 
-	cc := v["customer_certificate"].(map[string]interface{})
+	cc := v["customer_certificate"].([]interface{})
+	cc0 := cc[0].(map[string]interface{})
 
-	certificateId, err := keyvault.ParseNestedItemID(cc["key_vault_certificate_id"].(string), keyvault.VersionTypeAny, keyvault.NestedItemTypeCertificate)
+	certificateId, err := keyvault.ParseNestedItemID(cc0["key_vault_certificate_id"].(string), keyvault.VersionTypeAny, keyvault.NestedItemTypeCertificate)
 	if err != nil {
 		return nil, err
 	}

